embedded/store/key_reader.go

```diff
@@ -33,25 +34,31 @@ type Snapshot struct {
 
 type valueRefInterceptor func(key []byte, valRef ValueRef) ValueRef
 
-// filter out entries when filter evaluates to true
-type FilterFn func(valRef ValueRef, t time.Time) bool
+// filter out entries when filter evaluates to a non-nil error
+type FilterFn func(valRef ValueRef, t time.Time) error
```

Comment:
For future: Currently it's not needed but shouldn't we plan to also support errors when filtering happens? i.e. if we check something on disk and read fails. That would return an error too, but it would not indicate that the entry should be filtered our but rather that there was an error determining if it should be filtered out.

This would be important in line 366: https://github.com/codenotary/immudb/pull/1168/files#diff-c632c3c2ae2f4bdfbf8378e41bdcfecc38a5a87142a5a3ba769d721e4a0eb4cbR366 - where getting any error from filter means the entry is skipped where we should skip it only if the error indicates that the entry should be skipped.